src/sql/workbench/api/common/extHostNotebook.ts

```diff
@@ -283,37 +284,35 @@ export class ExtHostNotebook implements ExtHostNotebookShape {
 		return matchingAdapters;
 	}
 
-	private findSerializationManagerForUri(uriString: string): SerializationManagerAdapter {
-		for (let manager of this.getAdapters(SerializationManagerAdapter)) {
-			if (manager.uriString === uriString) {
-				return manager;
-			}
-		}
-		return undefined;
-	}
-
-	private findExecuteManagerForUri(uriString: string): ExecuteManagerAdapter {
+	private findExecuteManagersForUri(uriString: string): ExecuteManagerAdapter[] {
+		let managers = [];
 		for (let manager of this.getAdapters(ExecuteManagerAdapter)) {
```

Comment:
`return this.getAdapters(ExecuteManagerAdapter).filter(manager => manager.uriString === uriString)`

Reply:
Do you need to check the provider ID like you do for the serialization managers?

Reply:
This function only gets used for handling cleanup when a notebook is closed, so we just need to get all the managers associated with that document URI.